app/src/pages/ExploreDataPage.tsx

```diff
@@ -33,85 +30,66 @@ import {
+function getPhraseValue(madLib: MadLib, segmentIndex: number): string {
+  const segment = madLib.phrase[segmentIndex];
   return typeof segment === "string"
     ? segment
-    : segment[phraseSelections[segmentIndex]];
+    : segment[madLib.activeSelections[segmentIndex]];
 }
 
-function ReportWrapper(props: {
-  madLibIndex: number;
-  phraseSelections: PhraseSelections;
-}) {
-  const madlib = MADLIB_LIST[props.madLibIndex];
-  switch (props.madLibIndex) {
+function ReportWrapper(props: { madLib: MadLib }) {
+  switch (props.madLib.index) {
     case 0:
-      return (
-        <DemoReport madlib={madlib} phraseSelections={props.phraseSelections} />
-      );
+      return <DemoReport madLib={props.madLib} />;
     case 1:
-      return (
-        <TellMeAboutReport
-          madlib={madlib}
-          phraseSelections={props.phraseSelections}
-        />
-      );
+      return <TellMeAboutReport madLib={props.madLib} />;
     case 2:
       return (
         <CompareStatesForVariableReport
-          state1={getPhraseValue(madlib, 3, props.phraseSelections)}
-          state2={getPhraseValue(madlib, 5, props.phraseSelections)}
-          variable={getPhraseValue(madlib, 1, props.phraseSelections)}
+          state1={getPhraseValue(props.madLib, 3)}
+          state2={getPhraseValue(props.madLib, 5)}
+          variable={getPhraseValue(props.madLib, 1)}
         />
       );
     case 3:
-      return <TimeReport />;
-
+      return <ChartDumpReport />;
     default:
       return <p>Report not found</p>;
   }
 }
 
 function ExploreDataPage() {
-  const [shareModalOpen, setShareModalOpen] = React.useState(false);
+  const [shareModalOpen, setShareModalOpen] = useState(false);
+
   const params = useSearchParams();
   useEffect(() => {
     // TODO - it would be nice to have the params stay and update when selections are made
     // Until then, it's best to just clear them so they can't become mismatched
     clearSearchParams([MADLIB_PHRASE_PARAM, MADLIB_SELECTIONS_PARAM]);
   }, []);
-  const [madLibIndex, setMadLibIndex] = useState(
-    Number(params[MADLIB_PHRASE_PARAM]) | 0
-  );
 
-  let defaultValuesWithOverrides = MADLIB_LIST[madLibIndex].defaultSelections;
+  const initalIndex = Number(params[MADLIB_PHRASE_PARAM]) | 0;
+  let defaultValuesWithOverrides = MADLIB_LIST[initalIndex].defaultSelections;
   if (params[MADLIB_SELECTIONS_PARAM]) {
     params[MADLIB_SELECTIONS_PARAM].split(",").forEach((override) => {
       const [key, value] = override.split(":");
       // Validate that key is in valid range
-      if (!Object.keys(MADLIB_LIST[madLibIndex].phrase).includes(key)) return;
+      if (!Object.keys(MADLIB_LIST[initalIndex].phrase).includes(key)) return;
       // Validate that value is in valid range
       if (
-        !Object.keys(MADLIB_LIST[madLibIndex].phrase[Number(key)]).includes(
+        !Object.keys(MADLIB_LIST[initalIndex].phrase[Number(key)]).includes(
           value
         )
       )
         return;
```

Comment:
While you're here, can you add curly braces around the return statements? I'm not sure if there's anything actually broken here, but I feel with javascript it's better to avoid if statements without curly braces because auto-formatting + semicolon insertion makes it really easy to accidentally break things